File: Demos/C#/TomatoQuest/GameLogic.cs
using System;
using System.Drawing;
using System.Collections.Generic;



using SwinGame;
using Graphics = SwinGame.Graphics;
using Bitmap = SwinGame.Bitmap;
using Font = SwinGame.Font;
using FontStyle = SwinGame.FontStyle;
using Event = SwinGame.Event;
using CollisionSide = SwinGame.CollisionSide;
using Sprite = SwinGame.Sprite;

using GameResources;

namespace TomatoQuest
{
    public static class GameLogic
    {
        public static void RunGame()
        {
            //Open a new Graphics Window
            Core.OpenGraphicsWindow("The Legend of the Tomato Quest", 640, 480);
            //Open Audio Device
            Audio.OpenAudio();
            //Load Resources
				
            Resources.LoadResources();

				Audio.PlayMusic(Resources.GameMusic("Theme"));

				Core.ToggleFullScreen();

            //Creates the Game Object
            Game _Game = new Game();

            StartGame.DisplayStartGameMessage();

            //Game Loop
            do
            {
                //Clears the Screen to Black
                Graphics.ClearScreen();

                //Runs the Game
                _Game.Run();

                //Refreshes the Screen and Processes Input Events
                Core.RefreshScreen(65);
                Core.ProcessEvents();

            } while (!Core.WindowCloseRequested() && !EndGame.HasGameEnded);

            EndGame.DisplayEndGameMessage();

            //Free Resources and Close Audio, to end the program.
            Resources.FreeResources();
            Audio.CloseAudio();
        }
    }
}

File: SDKs/Base/DOTNet/VB/GameLogic.vb
Module GameLogic
    Public Sub Main()
        'Opens a new Graphics Window
        Core.OpenGraphicsWindow("Game", 800, 600)

        'Open Audio Device
        Audio.OpenAudio()

        'Load Resources
        LoadResources()

        'Game Loop
        Do
            'Clears the Screen to Black
            SwinGame.Graphics.ClearScreen()

            Graphics.FillRectangle(Color.Red, 20, 200, 200, 100)
            Graphics.FillRectangle(Color.Green, 220, 200, 200, 100)
            Graphics.FillRectangle(Color.Blue, 420, 200, 200, 100)

            Text.DrawText("Hello World", Color.Red, GameFont("Courier"), 20, 310)
            Text.DrawText("Hello World", Color.Green, GameFont("Courier"), 220, 310)
            Text.DrawText("Hello World", Color.Blue, GameFont("Courier"), 420, 310)

            Text.DrawFramerate(0, 0, GameFont("Courier"))
            Text.DrawText("Hello World", Color.White, GameFont("ArialLarge"), 50, 50)

            'Refreshes the Screen and Processes Input Events
            Core.RefreshScreen()
            Core.ProcessEvents()

        Loop Until SwinGame.Core.WindowCloseRequested() = True

        'Free Resources and Close Audio, to end the program.
        FreeResources()
        Audio.CloseAudio()

    End Sub

End Module
